File: Android/Kotlin/DogListDemo/app/src/main/java/com/suavepirate/doglistdemo/DogItemViewHolder.kt
package com.suavepirate.doglistdemo

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

class DogItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val nameTextView: TextView = itemView.findViewById(R.id.nameLabel)
    val ageTextView: TextView = itemView.findViewById(R.id.ageLabel)
    val adoptedTextView: TextView = itemView.findViewById(R.id.adoptedLabel)

    fun update(dog: AdoptableDog) {
        nameTextView.text = dog.name
        ageTextView.text = "${dog.age} years old"
        adoptedTextView.text = if(dog.isAdopted) "Adopted" else "Available"
    }
}

File: Android/CSharp/DogListDemo/DogItemViewHolder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Support.V7.Widget;
using Android.Views;
using Android.Widget;
using DogListDemo.Models;

namespace DogListDemo
{
    public class DogItemViewHolder : RecyclerView.ViewHolder
    {
        public TextView NameTextView { get; set; }
        public TextView AgeTextView { get; set; }
        public TextView AdoptedTextView { get; set; }
        public DogItemViewHolder(View itemView) : base(itemView)
        {
            NameTextView = itemView.FindViewById<TextView>(Resource.Id.NameLabel);
            AgeTextView = itemView.FindViewById<TextView>(Resource.Id.AgeLabel);
            AdoptedTextView = itemView.FindViewById<TextView>(Resource.Id.AdoptedLabel);
        }

        public void Update(AdoptableDog dog)
        {
            NameTextView.Text = dog.Name;
            AgeTextView.Text = $"{dog.Age} years old";
            AdoptedTextView.Text = dog.IsAdopted ? "Adopted" : "Available";
        }
    }
}